### PR TITLE
scan: merge decoder code and add schema to oracle script bridge code tab

### LINKED ISSUE #2031: Merge Params Decoder and Results Decoder in single file in Oracle Script Bridge

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #2031, #2032

### CODE REVIEW

File: scan/__tests__/Obi_test.re [hunk 1]
```diff
@@ -212,6 +228,25 @@ library ParamsDecoder {
...
+        internal
+        pure
+        returns (Params memory result)
+    {
+        Obi.Data memory data = Obi.from(_data);
+        result.symbol = string(data.decodeBytes());
+        result.multiplier = data.decodeU64();
+    }
+}
+
```
Comment (evilpeach) [inline scan/__tests__/Obi_test.re]: I think it's duplicated with `should be able to generate solidity` test case.

File: scan/src/bindings/Obi.re [hunk 2]
```diff
@@ -217,6 +214,20 @@ library $(dataTypeString)Decoder {
...
+
+import "./Obi.sol";
+
+|j};
+  let paramsCodeOpt = generateDecodeLibSolidity(schema, Params);
+  let resultCodeOpt = generateDecodeLibSolidity(schema, Result);
+  switch (paramsCodeOpt, resultCodeOpt) {
+  | (Some(paramsCode), Some(resultCode)) => Some(template ++ paramsCode ++ resultCode)
+  | _ => None
+  };
```
Comment (evilpeach) [inline scan/src/bindings/Obi.re]: Let's use `Let%Opt`

File: scan/src/components/oracle-script/OracleScriptBridgeCode.re [hunk 3]
```diff
@@ -153,21 +153,27 @@ module LanguageIcon = {
...
+  | Solidity => "Decoders.sol"
   | Go => {j|$(dataTypeString)Decoder.go|j}
   };
 };
 
 let getCodeFromSchema = (~schema, ~language, ~dataType) => {
   switch (language) {
-  | Solidity => Obi.generateDecoderSolidity(schema, dataType)
+  | Solidity => Obi.generateDecoderSolidity(schema)
   | Go => Obi.generateDecoderGo("main", schema, dataType)
```
Comment (evilpeach) [inline scan/src/components/oracle-script/OracleScriptBridgeCode.re:164]: Could we remove `dataType` on Go too?

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Comment by evilpeach** [comment]
@tansawit Let's get this PR done by next Wednesday.

File: scan/src/components/oracle-script/OracleScriptBridgeCode.re (hunk 3 above)
Comment (tansawit) [reply scan/src/components/oracle-script/OracleScriptBridgeCode.re:164]: We do not have `ParamsDecoder` for Go yet. And I believe we agreed to leave it as is for now until IBC is available.

**Review by evilpeach — APPROVED** [review]
LGTM

**Pull request merged**